ui-performance.md

```diff
@@ -0,0 +1,470 @@
+  The debug build compiles the Dart code “just in time” (also called JIT)
+  as the app runs, but profile and release builds are pre-compiled to native
+  instructions (also called “ahead of time”, or AOT) before the app is
+  loaded onto the device.
+</aside>
+
+### Compile in profile mode
+
+Flutter’s profile mode compiles and launches your application almost
+identical to release mode, but with just enough additional functionality
+to allow debugging performance problems.
+For example, profile mode provides tracing information to
+[Observatory](https://dart-lang.github.io/observatory/) and other tools.
+
+Launch the app in profile mode as follows:
+
+<ul markdown="1">
+<li markdown="1">In Android Studio, use the **Run > Profile…** menu item.
+</li>
+<li markdown="1">In IntelliJ, use the **Run > Flutter Run** in **Profile
+    Mode** menu item.
+</li>
+<li markdown="1"> In VSCode ... [PENDING]
+</li>
+<li markdown="1">From the command line, use the `--profile` flag:
+{% prettify sh %}
+$ flutter run --profile
+{% endprettify %}
+</li>
+</ul>
+
+For more information on how the different modes work, see [Flutter's
+modes](https://github.com/flutter/flutter/wiki/Flutter%27s-modes).
+
+### Use the performance overlay
+
+The performance overly displays performance statistics in two graphs
+that show you where time is being spent in your app.
+If the UI is janky (skipping frames), these graphs help you figure out why.
+
+The graphs display on top of your running app, but they aren’t drawn
+like a normal widget—the Flutter engine itself paints the overlay.
+
+The next section describes how to enable the performance overlay,
+and use it it to diagnose the cause of any jank in your application.
+
+## The performance overlay
+
+<div class="whats-the-point" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* Turn on the overlay with the Flutter inspector, at the command
+  line, or programmatically.
+* Observe the app's performance graphically.
+* One graph displays performance of Dart code. The other displays
+  performance of the graphics engine.
+* The overlay displays the last 300 frames for each engine.
+
+</div>
+
+The
+[PerformanceOverlay](https://docs.flutter.io/flutter/widgets/PerformanceOverlay-class.html)
+widget, from the
+[widget library](https://docs.flutter.io/flutter/widgets/widgets-library.html),
+displays performance statistics using two graphs.
+The following screenshot shows the performance overlay running on the Flutter
+Gallery example:
+
+<center><img src="images/performance-overlay-green.png" alt="screenshot of performance overlay showing zero jank"></center>
+<center>The graphs produced by the performance overlay.<br>THe vertical green bar represents the current frame.</center>
+
+Each graph represents the last 300 frames for the following two threads used in every Flutter
+app:
+
+* Engine<br>
```

Comment:
These names were confusing (since the `Engine` label really mainly pointed to code running in the framework). We renamed them UI and GPU instead to match the threads.